.github/workflows/tox.yml

```
@@ -238,3 +238,52 @@ jobs:
         uses: codecov/codecov-action@015f24e6818733317a2da2edd6290ab26238649a  # v5.0.7
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
+
+      - if: ${{ (success() || failure()) && contains(matrix.coverage, 'github') && matrix.pytest == 'true' }}
+        name: add run info to coverage filename
+        run: |
+          find ${{ github.workspace }} -name .coverage
+          mv $(find ${{ github.workspace }} -name .coverage) .coverage.${{ github.sha }}-${{ runner.os }}-${{ runner.arch }}-${{ matrix.toxenv }}
+        shell: sh
+
+      - name: Upload coverage data to GitHub
+        if: ${{ (success() || failure()) && contains(matrix.coverage, 'github') && matrix.pytest == 'true' }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: .coverage.${{ github.sha }}-${{ runner.os }}-${{ runner.arch }}-${{ matrix.toxenv }}
+          path: .coverage.${{ github.sha }}-${{ runner.os }}-${{ runner.arch }}-${{ matrix.toxenv }}
```

Ok I am getting desperate now:

```suggestion

      - name: Upload coverage data to GitHub
        if: ${{ (success() || failure()) && contains(matrix.coverage, 'github') && matrix.pytest == 'true' }}
        uses: actions/upload-artifact@v4
        with:
          name: .coverage.${{ github.sha }}-${{ runner.os }}-${{ runner.arch }}-${{ matrix.toxenv }}
          path: **/.coverage
```